File: oop/py/sort_dates.py
#!/usr/bin/python

import sys

## change this path to where Date.py is defined.
load_path = '/home/vladimir/programming/python/oop/'

if __name__ == '__main__':
    if not load_path in sys.path:
        sys.path.append(load_path)

from Date import Date

dates = []
dates.append(Date(11, 19, 2015))
dates.append(Date(11, 20, 2015))
dates.append(Date(11, 21, 2015))

dates.sort(key=lambda d: d.getDay())
for d in dates: print d.toMDYString()

File: oop/pl/sort_dates.pl
#!/usr/bin/perl

## change this path to where Date.pm is defined
use lib '/home/vladimir/programming/pl/oop/';
use Date;
use strict;
use warnings;

my @dates = ();

push(@dates, Date->make(11, 19, 2015));
push(@dates, Date->make(11, 20, 2015));
push(@dates, Date->make(11, 21, 2015));

## sort dates by day
print $_->toMDYString() . "\n" foreach(sort { $a->getDay() <=> $b->getDay() } @dates);
